server/src/controllers/authentication.ts

```diff
@@ -19,4 +20,26 @@ const login = async (req: Request, res: Response) => {
   res.cookie('token', token);
   return { status: 200, msg: 'logged in successfully' };
 };
-export default login;
+const freelancerSignUp = async (req: Request, res: Response) => {
+  const {
+    Title, Major, portfolio, description, image, id,
+  } = req.body;
+  const user: UserInstance | null = await User.findOne({ where: { id } });
+  if (!user) throw serverErrs.BAD_REQUEST('user id doen\'t exist');
+  const { name, role } = user;
+  const token = await generateToken({ userID: id, role, name });
+
+  await freelancerValidate.validate(req.body);
+
+  const freelancer = await Freelancer.create({
+    image,
+    title: Title,
+    major: Major,
+    brief: description,
+    userId: id,
+    portfolio,
+  });
+  res.cookie('token', token);
+  return { status: 200, data: freelancer, msg: 'successful login' };
```

Comment:
```suggestion
  return { status: 201, data: freelancer, msg: 'successful login' };
```
And I think we don't need the whole data `name`, `role`, and freelancer `id` would be enough